src/library.rs

```diff
@@ -445,6 +445,28 @@ pub struct Record {
     pub disguised: bool,
 }
 
+impl Record {
+    pub fn has_free(&self) -> bool {
+        self.functions.iter().any(|f| f.name == "free") || (self.is_copy() && self.has_destroy())
+    }
+
+    pub fn is_copy(&self) -> bool {
```

Comment:
```suggestion
    pub fn has_copy(&self) -> bool {
```

If it has a `copy` function then the type does not implement the `Copy` trait but only `Clone`.

There's also the `copy_into` function btw

Reply:
> There's also the copy_into function btw

Should we check for that as well when checking the `has_copy` ?

Reply:
Yes. If either exists then the type can't be `Copy` (as copying is more than a plain `memcpy`)

Reply:
Done, and after adding the check for the copy type. It no longer generates `&[gdk::Atom]` but instead it generates `&[&gdk::Atom]` which makes the gtk3-rs change unneeded.

Reply:
Ah why? `gdk::Atom` is `Copy`

Reply:
Because it doesn't provide a `copy` function, maybe we need to check for something else as well?

Reply:
Yes, because it does **not** provide a `copy` / `copy_into` function it implements `Copy` (i.e. copying/cloning it is just a `memcpy` and nothing else)